### PR TITLE
[SEASK-55] Extension sample data for Grandbend 

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+DECLARE @localEducationAgencyId INT;
+DECLARE @student604822USI INT;
+DECLARE @student604823USI INT;
+DECLARE @student604822EntryDate date;
+DECLARE @student604823EntryDate date;
+
+SELECT @grandBendHighSchoolId = SchoolId , @localEducationAgencyId= localEducationAgencyId FROM edfi.School WHERE SchoolId = 255901001
+SELECT @student604822USI = StudentUSI FROM edfi.Student WHERE StudentUniqueId = '604822'
+SELECT @student604823USI = StudentUSI FROM edfi.Student WHERE StudentUniqueId = '604823'
+SELECT @student604822EntryDate= EntryDate FROM edfi.[StudentSchoolAssociation] where StudentUSI =@student604822USI
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql]:
```suggestion
SELECT @student604822EntryDate = EntryDate FROM edfi.[StudentSchoolAssociation] WHERE StudentUSI = @student604822USI
```

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+SELECT @student604823EntryDate =EntryDate FROM edfi.[StudentSchoolAssociation] where StudentUSI =@student604823USI
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql]:
```suggestion
SELECT @student604823EntryDate = EntryDate FROM edfi.[StudentSchoolAssociation] WHERE StudentUSI = @student604823USI
```

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+DECLARE @grandBendHighSchoolId INT;
+DECLARE @localEducationAgencyId INT;
+DECLARE @student604822USI INT;
+DECLARE @student604823USI INT;
+DECLARE @student604822EntryDate date;
+DECLARE @student604823EntryDate date;
+
+SELECT @grandBendHighSchoolId = SchoolId , @localEducationAgencyId= localEducationAgencyId FROM edfi.School WHERE SchoolId = 255901001
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql]:
```suggestion
SELECT @grandBendHighSchoolId = SchoolId , @localEducationAgencyId = localEducationAgencyId FROM edfi.School WHERE SchoolId = 255901001
```

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/PgSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+declare student604822EntryDate date;
+declare student604823EntryDate date;
+
+begin
+
+SELECT  SchoolId ,localEducationAgencyId into grandBendHighSchoolId ,residentLocalEducationAgencyId FROM edfi.School WHERE SchoolId = 255901001;
+SELECT StudentUSI into student604822USI  FROM edfi.Student WHERE StudentUniqueId = '604822';
+SELECT StudentUSI into student604823USI FROM edfi.Student WHERE StudentUniqueId = '604823';
+SELECT EntryDate into student604822EntryDate FROM edfi.StudentSchoolAssociation where StudentUSI = student604822USI;
+SELECT EntryDate into student604823EntryDate FROM edfi.StudentSchoolAssociation where StudentUSI = student604823USI;
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/PgSql/Data/Ods/0010-SampleExtensionsData.sql]:
```suggestion
SELECT EntryDate into student604822EntryDate FROM edfi.StudentSchoolAssociation WHERE StudentUSI = student604822USI;
SELECT EntryDate into student604823EntryDate FROM edfi.StudentSchoolAssociation WHERE StudentUSI = student604823USI;
```

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/PgSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+declare residentLocalEducationAgencyId int;
+declare student604822USI int;
+declare student604823USI int;
+declare student604822EntryDate date;
+declare student604823EntryDate date;
+
+begin
+
+SELECT  SchoolId ,localEducationAgencyId into grandBendHighSchoolId ,residentLocalEducationAgencyId FROM edfi.School WHERE SchoolId = 255901001;
+SELECT StudentUSI into student604822USI  FROM edfi.Student WHERE StudentUniqueId = '604822';
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/PgSql/Data/Ods/0010-SampleExtensionsData.sql]:
```suggestion
SELECT StudentUSI into student604822USI FROM edfi.Student WHERE StudentUniqueId = '604822';
```

**Review by blmeyers — DISMISSED** [review]
The changes look fine, but I'd like to see a successful template build of this branch that is actively running these scripts.  If you look at https://intedfitools1.msdf.org/buildConfiguration/OdsPlatform_OdsApi_Packages_EdFiOdsPopulatedTemplateSk/130657?buildTab=log&focusLine=3&linesState=1994.2148.2416 you'll see its not running these scripts, because it seems to be locked to a specific plugin version.  Let's update the plugin version so it can use these scripts, and make sure the template is still building successfully, and that we can retrieve these entities from this template.

File: sample-data/plugin/configuration.packages.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
   "packages": {
     "sk": {
       "PackageName": "EdFi.Ods.Extensions.Sk",
-      "PackageVersion": "1.0.46",
+      "PackageVersion": "1.0.150",
```
Comment (vimayya) [inline sample-data/plugin/configuration.packages.json]: @semalaiappan https://github.com/Ed-Fi-Alliance-OSS/Starter-Kit-SEA-Data-Collection/blob/main/packer/scripts/sk.ps1#L15 needs to be updated as well. Currently this package version is in two places.

Comment (blmeyers) [reply sample-data/plugin/configuration.packages.json]: Yeah, this is looking good at load time and is now correctly running the script as expected.  Let's get it caught up with latest and do the update Vinaya mentioned and then we should be good to get it in.

Comment (semalaiappan) [reply sample-data/plugin/configuration.packages.json]: @vimayya @blmeyers   I updated package version in packer folder also

File: sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql [hunk 7]
```diff
@@ -3,33 +3,11 @@
...
-
-INSERT INTO [sk].[StudentSchoolAssociationExtension]
-([EntryDate],[SchoolId],[StudentUSI],[ResidentLocalEducationAgencyId],[ResidentSchoolId],[ReportingSchoolId])
-VALUES
-(@student604823EntryDate,@grandBendHighSchoolId,@student604823USI,@localEducationAgencyId,@grandBendHighSchoolId,@grandBendHighSchoolId)
+INSERT INTO sk.StudentSchoolAssociationExtension (EntryDate, SchoolId, StudentUSI, ResidentLocalEducationAgencyId, ResidentSchoolId, ReportingSchoolId)
+SELECT ssa.EntryDate, ssa.EducationOrganizationId, st.StudentUSI, sc.LocalEducationAgencyId, sc.SchoolId, sc.SchoolId
+FROM edfi.Student st
+JOIN edfi.StudentSchoolAssociation ssa ON st.StudentUSI = ssa.StudentUSI
+JOIN edfi.School sc ON ssa.EducationOrganizationId = sc.SchoolId
```
Comment (blmeyers) [inline sample-extension/EdFi.Ods.Extensions.Sk/Artifacts/MsSql/Data/Ods/0010-SampleExtensionsData.sql]: This is joining the EducationOrganizationId (sourced from the optional Graduation Plan) to the SchoolId, when there's no guarantee that we'll have a GraduationPlan.EducationOrganizationId, or that it has to be school level.  Is the intent here for this to be joining SchoolId to SchoolId?

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**